src/create.cpp

```diff
@@ -40,25 +41,49 @@ unsigned Create::run()
         throw rest::Error(ret.error());
     }
 
-    std::string assetJson = m_request.body();
-
-    auto ret = AssetManager::createAsset(assetJson, user.login());
-    if (!ret) {
-        auditError(ret.error());
-        throw rest::errors::Internal(ret.error());
+    cxxtools::SerializationInfo si;
+    try {
+        std::stringstream          jsonIn(m_request.body());
+        cxxtools::JsonDeserializer deserializer(jsonIn);
+        deserializer.deserialize(si);
+    } catch (const std::exception& e) {
+        logError(e.what());
+        throw rest::errors::Internal(e.what());
     }
 
-    auto createdAsset = AssetManager::getItem(*ret);
-
-    if (!createdAsset) {
-        auditError(createdAsset.error());
-        throw rest::errors::Internal(createdAsset.error());        
-    } 
+    cxxtools::SerializationInfo assetsJsonList;
+    if (si.findMember("assets")) {
+        assetsJsonList = si.getMember("assets");
+    } else {
+        assetsJsonList.addMember(m_request.body());
```

Comment:
strange, i didn't catch it, body is already parsed, but you adding to list.. mmm.... string? is it work?